### DISCUSSION TITLE
FlxG.collide sometimes doesn't work

### DISCUSSION THREAD

**Discussion by TheHENOOB** [HaxeFlixel/flixel#2327] [Q&A] [answered]
First i will show the code
```
FlxG.collide(_player, _bg);
FlxG.collide(_enemygroup, _bg);
FlxG.collide(_player, _enemygroup, endgame);
```
_bg is a FlxTypedGroup of FlxSprites that has two FlxShapeBox on the up and the down side of the screen and also a non-touchable sprite on the middle of the screen only to be used to display animation
![image](https://user-images.githubusercontent.com/74883505/122842708-43f59600-d2d4-11eb-8a9c-4bcc9f1ea59e.png)

_enemygroup is another FlxTypedGroup but this time using a class extending FlxSprite called Enemy which generates a amount of enemys on the right side of the screen

the situation is that _player is not colliding with _bg members and when _player collides with a member of _enemygroup in the left and right side, the member of _enemygroup stops moving and the endgame callback doesn't get called

i fix this problem by
```
if (_player.y <= 34)
{
	_player.y = 34;
}
else if (_player.y >= 660)
{
	_player.y = 660;
}
```
which are the y position on where the player should stop moving by the two boxes and
```
if (_enemy.velocity.x == 0)
{
	endgame();
	_enemy.velocity.x = _enemy.SPEED;
}
```
since a member of _enemygroup stops moving after colliding with the player

I'm trying to find a way to use FlxG.collide normally without those glitches happening and additional code, any help i appreciate

**Comment by RichardBray** [comment, marked as answer]
I've had better experiences using `FlxTypedSpriteGroup ` over `FlxTypedGroup<FlxSprite>`
https://api.haxeflixel.com/flixel/group/FlxTypedSpriteGroup.html

Although, have you tried adding the collision in the update function of the FlxTypedGroup class?

**Reply by TheHENOOB** [reply]
> I've had better experiences using FlxTypedSpriteGroup over FlxTypedGroup<FlxSprite>

I tried using this instead of FlxTypedGroup for both _bg and _enemygroup and the collision still doesn't work

> Although, have you tried adding the collision in the update function of the FlxTypedGroup class?

I tried with _bg class, no success

other notes:

- I tried placing the FlxG.collide above and below super.update()
- I don't know if flixel has a collision limit

EDIT: Deleted the third note because i checked it by commenting the forEach line, no success

**Reply by TheHENOOB** [reply]
If possible i can temporarily open-source my game and check what is wrong with my code

**Reply by RichardBray** [reply]
Yeah I think that would be helpful 👍

**Reply by TheHENOOB** [reply]
here it is: https://github.com/TheHENOOB/CarTripClone

**Reply by TheHENOOB** [reply]
Any updates? Is it a bug or a mistake i made on the code?

**Reply by TheHENOOB** [reply]
I fixed one of the issues and it is located on the Enemy.hx class
```
if (!FlxG.collide())
{
	velocity.y = 0;
}
```
Removing those lines of code, the player can successfully collide the enemygroup and call the callback, i think if i have to add FlxG.collide on a FlxSprite class, i have to do it in a other way

PS: Ill delete the CarTripClone Repository when i find out how to fix the _bg class

**Reply by TheHENOOB** [reply]
Problem Solved:

On the player class immovable was set to true, meaning that no other class would collide with the class specially _bg, i will now delete CarTripClone Repository now

**Reply by RichardBray** [reply]
Ah, I'm glad you figured it out. Ran your game on my machine and tried to figure it out but wasn't able to. The game looks really cool by the way 👍
